qtawesome/iconic_font.py

```diff
@@ -450,3 +495,62 @@ def _icon_by_painter(self, painter, options):
         """Return the icon corresponding to the given painter."""
         engine = CharIconEngine(self, painter, options)
         return QIcon(engine)
+
+    def _get_fonts_directory(self):
+        """
+        Get bundled fonts directory.
+
+        On Windows an attempt to install the fonts per user is done
+        to prevent errors with fonts loading.
+        See spyder-ide/qtawesome#167 and spyder-ide/spyder#18642
+        """
+        fonts_directory = os.path.join(
+            os.path.dirname(os.path.realpath(__file__)), 'fonts')
+        if os.name == 'nt':
+            fonts_directory = self._install_fonts(fonts_directory)
+        return fonts_directory
+
+    def _install_fonts(self, fonts_directory):
+        """Copy the fonts to the user Fonts folder."""
+        user_fonts_dir = os.path.join(
+            os.environ['LOCALAPPDATA'], 'Microsoft', 'Windows', 'Fonts')
```

Comment:
Is `LOCALAPPDATA` expected to always be set? If that's not the case, perhaps we should validate it before proceeding so that Spyder doesn't crash at startup.

Reply:
I think is always set unless you launch things programatically and set a specific env like for example the process Spyder uses to launch Pylint, but I will add a validation just in case :+1: